datadog_lambda/tracing.py

```diff
@@ -87,24 +88,26 @@ def _convert_xray_sampling(xray_sampled):
 def _get_xray_trace_context():
     if not is_lambda_context():
         return None
-
+    #For review, I dont want to edit parse... as we still need to _convert... in this file
     xray_trace_entity = parse_xray_header(
         os.environ.get(XrayDaemon.XRAY_TRACE_ID_HEADER_NAME, "")
     )
     if xray_trace_entity is None:
         return None
-    trace_context = {
-        "trace-id": _convert_xray_trace_id(xray_trace_entity.get("trace_id")),
-        "parent-id": _convert_xray_entity_id(xray_trace_entity.get("parent_id")),
-        "sampling-priority": _convert_xray_sampling(xray_trace_entity.get("sampled")),
-    }
+    trace_context = Context(
+        trace_id= _convert_xray_trace_id(xray_trace_entity.trace_id),
+        span_id=_convert_xray_entity_id(xray_trace_entity.span_id),
+        sampling_priority=_convert_xray_sampling(xray_trace_entity.sampling_priority),
+        dd_origin=xray_trace_entity.dd_origin
+    )
     logger.debug(
         "Converted trace context %s from X-Ray segment %s",
         trace_context,
         (
-            xray_trace_entity["trace_id"],
-            xray_trace_entity["parent_id"],
-            xray_trace_entity["sampled"],
+            xray_trace_entity.trace_id,
+            xray_trace_entity.span_id,
+            xray_trace_entity.sampling_priority,
+            xray_trace_entity.dd_origin,
         ),
```

Comment:
TODO: Try to avoid allocating this extra tuple.

Reply:
The `xray_trace_entity` will already be printed when the `trace_context` is printed.